external_parser/event_processors/joined_event.h

```diff
@@ -88,9 +88,8 @@ struct cb_joined_event : public typed_joined_event {
 
     int index = interaction_data.actions[0];
     auto action = interaction_data.actions[0];
-    auto probability = interaction_data.probabilities[0] *
-                       (1.f - interaction_data.probabilityOfDrop);
-    auto weight = 1.f - interaction_data.probabilityOfDrop;
+    auto probability = interaction_data.probabilities[0];
```

Comment:
check if probabilityOfDrop is 1.f and if so maybe log and set weight to zero?

Reply:
hm, not 100% sure if setting weight to zero is safe. Also, dsjson parser is considering that kind of examples invalid and got skipped. Maybe it is better to replicate this behavior here for consistency and explore zero weight as next step?

Reply:
Consistency is good, that would require `fill_in_label` to indicate back to joiner whether it should abort this example or not

Reply:
q: does this probability need to multiply by (1-pdrop) ?

Reply:
Why do we need to take it into account both in probability and weight? 
Eventually all of them are going to the cost and in some cases (probably in ips) we should see weight / probability multiplier which probably can be mimicked by setting weight to 1 and multiplying probability by (1-pdrop), but not sure why we should do it this way.

Reply:
I think setting one or the other has an effect on reported average loss, comparing results with dsjson parser

Reply:
looking at this 2 things need to be done here 
1. looks like for dsjson we set the weight for **every example** (see [here](https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/parse_example_json.h#L1664) so we should do the same here too
2. @cheng-tan is right, we need to apply the pdrop to the probability too as it was before (that's how it is set during dsjson joining and before the dsjson format arrives to vw for parsing)

Reply:
1. could be done in a separate PR and should also be applied to ccb/cats

Reply:
1. agree, good catch!
2. let's discuss offline

Reply:
1. and agree, let me fix it in another PR - that one seems big enough

Reply:
2. discussed offline - looks like the change was propagated from the legacy backend code which should be fixed.